src/workers/search/get-matches.js

```diff
@@ -20,8 +20,17 @@ export default function getMatches(
   for (let i = 0; i < lines.length; i++) {
     let singleMatch;
     const line = lines[i];
+    let previousLastIndex = 0;
     while ((singleMatch = regexQuery.exec(line)) !== null) {
       matchedLocations.push({ line: i, ch: singleMatch.index });
+
+      // exec() is supposed to update the lastIndex property but sometimes
+      // doesn't and can cause this to be an infinite loop by matching on the
+      // same index.  A work around is to update the lastIndex manually
+      if (regexQuery.lastIndex === previousLastIndex) {
```

Comment:
Instead of saving `previousLastIndex` here, we can check `singleMatch[0] === ""` to keep things simpler.

Also, I think it's fine, but it might also be good to add an assertion here that `!regexQuery.unicode` since incrementing this index by a single codeunit could cause issues if the regex was trying to match in unicode mode.

If you're curious, I didn't think about the unicode issue either, but I looked at https://tc39.github.io/proposal-string-matchall/#%RegExpStringIteratorPrototype%.next since it has to deal with this same issue. You can see step `11.a.ii.2` it does the same type of thing.